internal/provider/resource_edge_cluster.go

```diff
@@ -0,0 +1,250 @@
+				Optional:    true,
+				Description: "Select whether Tier-1 being created per this spec is hosted on the new Edge cluster or not (default value is false, meaning hosted)",
+			},
+			"internal_transit_subnets": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Subnet addresses in CIDR notation that are used to assign addresses to logical links connecting service routers and distributed routers",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"transit_subnets": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Transit subnet addresses in CIDR notation that are used to assign addresses to logical links connecting Tier-0 and Tier-1s",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"edge_node": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "The nodes in the edge cluster",
+				Elem:        edge_cluster.EdgeNodeSchema(),
+			},
+		},
+	}
+}
+
+func resourceEdgeClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*api_client.SddcManagerClient).ApiClient
+
+	params := &nsxt_edge_clusters.CreateEdgeParams{
+		EdgeCreationSpec: edge_cluster.GetEdgeClusterCreationSpec(d),
+		Context:          ctx,
+	}
+
+	_, task, err := c.NSXTEdgeClusters.CreateEdge(params)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	tflog.Info(ctx, "Edge cluster creation has started.")
+	err = meta.(*api_client.SddcManagerClient).WaitForTaskComplete(ctx, task.Payload.ID, false)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var clusters *nsxt_edge_clusters.GetEdgeClustersOK
+	clusters, err = c.NSXTEdgeClusters.GetEdgeClusters(nil)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	for _, cluster := range clusters.Payload.Elements {
+		if cluster.Name == d.Get("name") {
+			d.SetId(cluster.ID)
+			tflog.Info(ctx, "Edge cluster created successfully.")
+			return nil
+		}
+	}
+
+	return diag.Errorf("Edge cluster creation failed - cluster not found in inventory")
+}
+
+func resourceEdgeClusterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*api_client.SddcManagerClient).ApiClient
+
+	_, err := getEdgeCluster(ctx, c, d.Id())
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceEdgeClusterDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	tflog.Info(ctx, "Edge cluster deletion is not implemented. See KB article 78635 for more information.")
+	return nil
+}
+
+func resourceEdgeClusterUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*api_client.SddcManagerClient).ApiClient
+
+	edgeClusterOk, err := getEdgeCluster(ctx, c, d.Id())
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if d.HasChange("edge_node") {
+		oldNodesChange, newNodesChange := d.GetChange("edge_node")
```

Comment:
maybe only oldNodes, newNodes. The arrays contain the old and new absolute values, right, not the deltas?

Reply:
Sure, renamed the varables